fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -2787,7 +2732,8 @@ public LoanTransaction handlePayDisbursementTransaction(final Long chargeId, fin
             final boolean isScheduleRegenerateRequired = isRepaymentScheduleRegenerationRequiredForDisbursement(actualDisbursementDate);
             this.actualDisbursementDate = null;
             this.disbursedBy = null;
-            boolean isDisbursedAmountChanged = !this.approvedPrincipal.equals(this.loanRepaymentScheduleDetail.getPrincipal());
+            boolean isDisbursedAmountChanged =
+                    !this.approvedPrincipal.equals(this.loanRepaymentScheduleDetail.getPrincipal().getAmount());
```

Comment:
this looks quite suspicious, to me... so we're comparing an `approvedPrincipal` with a `getPrincipal().getAmount()`- is that really correct? Could someone with a better functional understanding of this code review this?

Reply:
This came as a result of Incompatible Caparison rule violation Exception thrown by SpotBugs.
approvedPrincipal is of Type BigDecimal (Java Type) while loanRepaymentScheduleDetail.getPrincipal() returns type Money (Fineract type which does not extend BigDecimal, this is an object with various other properties). Will the JVM be able to compare BigDecimal and Money? I reasoned this wont work so I logically used the amount (of Type BigDecimal) to compare with approvedPrincipal. I might be missing something here. Advice. @vishwasbabu , @nazeer1100126 , @Nayan .

I could just also suppress this if its a false positive and my reasoning is wrong.

Reply:
I am putting this back and suppressing SpotBugs for this method. We can take a look at it later.